### PR TITLE
[quantization] Re-work Conv2d

### PR DESCRIPTION
**Pull request by jamesr66a**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24117 [quantization] JIT serialization for Conv2d
* #24116 [quantization] state_dict serialization for Conv2d + some bugfixes
* **#24115 [quantization] Re-work Conv2d**

Differential Revision: [D16765474](https://our.internmc.facebook.com/intern/diff/D16765474)

### CODE REVIEW

File: torch/nn/quantized/modules/conv.py [hunk 1]
```diff
@@ -49,80 +48,48 @@ class Conv2d(_ConvNd):
...
+        self.bias = torch._empty_affine_quantized([out_channels],
+                                                  scale=1, zero_point=0,
+                                                  dtype=torch.qint32)
+        self.scale = 1.0
+        self.zero_point = 0
 
-    @property
-    def zero_point(self):
-        return self._zero_point.item()
+    def set_weight(self, w):
```
Comment (zafartahirov) [inline torch/nn/quantized/modules/conv.py]: Is there an advantage of using explicit setters/getters instead of the properties? I feel like we are stepping off the common pattern, where weight/bias are just variables, not methods.

**Review by raghuramank100 — APPROVED** [review]

File: torch/nn/quantized/modules/conv.py (hunk 1 above)
Comment (dzhulgakov) [reply torch/nn/quantized/modules/conv.py]: are properties breaking with TorchScript? but I agree - would be nice to make them work (I think it should be possible to disambiguate properties and fields because properties are on the class level, not object level)

Comment (jamesr66a) [reply torch/nn/quantized/modules/conv.py]: Yes, properties are unsupported in TorchScript, I can file an issue
